ironfish/src/mining/pool.ts

```diff
@@ -337,7 +337,9 @@ export class MiningPool {
     Assert.isNotNull(this.currentHeadTimestamp)
     Assert.isNotNull(this.currentHeadDifficulty)
 
-    const latestBlock = this.miningRequestBlocks.get(this.nextMiningRequestId - 1)
+    const currentBlock = this.miningRequestBlocks.get(this.nextMiningRequestId - 1)
+    const latestBlock = JSON.parse(JSON.stringify(currentBlock)) as typeof currentBlock
```

Comment:
What do you think about the approach used here? https://github.com/iron-fish/ironfish/blob/master/ironfish/src/mining/pool.ts#L223-L224

```suggestion
    const latestBlock = Object.assign({}, currentBlock)
    latestBlock.header = Object.assign({}, currentBlock.header)
```

I suspect the JSON stringify and parse is slower than `Object.assign`, but it also makes me a bit nervous as it could introduce subtle serialization bugs

Reply:
Updated.